core/src/test/java/test/org/apache/spark/JavaAPISuite.java

```diff
@@ -541,7 +541,7 @@ public void treeAggregate() {
     JavaRDD<Integer> rdd = sc.parallelize(Arrays.asList(-5, -4, -3, -2, -1, 1, 2, 3, 4), 10);
     Function2<Integer, Integer, Integer> add = (a, b) -> a + b;
     for (int depth = 1; depth <= 10; depth++) {
-      int sum = rdd.treeAggregate(0, add, add, depth);
+      int sum = rdd.treeAggregate(0, add, add, depth, false);
```

Comment:
nit: already the default value is `false` right?

Reply:
theres an overloaded version, so we have to pass in `false` or else it does not compile

Reply:
The other one is not overloaded right, it has additional param `depth`. This seems to be java <-> scala interoperability issue which is why even with default param, the value needs to be specified I think. Not exactly the same case but similar problem discussed here - https://stackoverflow.com/questions/13059528/instantiate-a-scala-class-from-java-and-use-the-default-parameters-of-the-const

Reply:
Right. in the bytecode it is not optional as there is no such thing in the bytecode. For Java callers it's required; Scala compiler inserts the default value when compiling Scala code.